### PR TITLE
gunpey.cpp: Add save states, Fix init, Add version info from startup screen, Minor cleanups

### PR DESCRIPTION
**Pull request by cam900**
Reduce duplicates, Unnecessary variables and lines
Fix naming, Spacing
Use alpha_blend_r16 for blending functions
Use shorter/correct type values
Add notes

### CODE REVIEW

File: src/mame/drivers/gunpey.cpp [hunk 1]
```diff
@@ -234,17 +234,21 @@ struct huffman_node_s
 	int arg1_val;
 };
 
+static inline u32 get_vmem_addr(int x, int y)
+{
+	return (x & 0x7ff) | ((y & 0x7ff) << 11);
+}
```
Comment (cuavas) [inline src/mame/drivers/gunpey.cpp:240]: Please leave the two blank lines between sections, and use `constexpr` for things that should be `constexpr` (also `inline` implies `static` in C++, there’s no need for both).  Also, why are the inputs signed when the output is unsigned?

File: src/mame/drivers/gunpey.cpp [hunk 2]
```diff
@@ -235,16 +235,22 @@ struct huffman_node_s
 };
 
 
+constexpr u32 get_vmem_addr(u16 x, u16 y)
```
Comment (angelosa) [inline src/mame/drivers/gunpey.cpp:238]: `vmem` is ambiguous (virtual memory?), should really be `vram` like everything else in the state machine.

Comment (cam900) [reply src/mame/drivers/gunpey.cpp:238]: vmem means video memory, both video ROM and RAM.
Both has same addressing format(2048x2048 size framebuffer-like addressing).

Comment (angelosa) [reply src/mame/drivers/gunpey.cpp:238]: So be `get_video_addr[ess]`.

**Pull request merged**